<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="homework.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> PDS Mini Project 2 <br>
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

## Group Members Name
  
<div class="alert alert-warning">
<p style="font-family: Arial Black; font-size:1.4em;color:Purple;">  
1.  G. R. Priyavarshini
<div class="alert alert-warning">
<p style="font-family: Arial Black; font-size:1.4em;color:Purple;">    
2.  G. R. Tamothar
<div class="alert alert-warning">
<p style="font-family: Arial Black; font-size:1.4em;color:Purple;">    
3.  T. Uma Maheshwari

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings("ignore")
from scipy.stats import skew
from matplotlib.ticker import PercentFormatter


### 1. Create a dictionary named 'europe_dict' which has names of countries in europe, their capital and their population. (2 marks)
   

europe_dict = {
       "Country":["Spain","France","Germany","Norway"],
       "Capital":["Madrid","Paris","Berlin","Oslo"],
       "Population":[46.77,66.03,80.62,5.084]
}

In [2]:
europe_dict = {
    "Country": ["Spain", "France", "Germany", "Norway"],
    "Capital": ["Madrid", "Paris", "Berlin", "Oslo"],
    "Population": [46.77, 66.03, 80.62, 5.084]
}
europe_df = pd.DataFrame(europe_dict)

In [3]:
europe_df

Country Capital  Population
0    Spain  Madrid      46.770
1   France   Paris      66.030
2  Germany  Berlin      80.620
3   Norway    Oslo       5.084

#### Summary: 
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">Step 1: Creating a dictionary 
    
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">Step 2: Passing dictionary inside a Dataframe (pd.DataFrame(europe_dict))

### 2. Use the data from question 1 and add a new column 'FamousFor'. The column should indicate what each country is famous for. (2 marks)

##### METHOD 1: ADDING A NEW COLUMN TO DATA FRAME USING insert

In [4]:
europe_df3=europe_df.copy(deep=True)
europe_df3.insert(3, "Famousfor", ["Wine","Perfume","Automobiles","Viking Heritage"],True)
europe_df3

Country Capital  Population        Famousfor
0    Spain  Madrid      46.770             Wine
1   France   Paris      66.030          Perfume
2  Germany  Berlin      80.620      Automobiles
3   Norway    Oslo       5.084  Viking Heritage

<strong>Summary: </strong> 
<div class="alert alert-danger"> 
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
It gives the freedom to add a column at any position we like and not just at the end. It also provides different options for inserting the column values.

##### METHOD 2: ADDING A NEW COLUMN TO DATA FRAME USING merge

In [5]:
# type your data here
FamousFor = {
    "Country": ["Spain", "France", "Germany", "Norway"],
    "FamousFor" : ["Wine","Perfume","Automobiles","Viking Heritage"]
}


In [6]:
df_FamousFor = pd.DataFrame(FamousFor)
df_FamousFor

Country        FamousFor
0    Spain             Wine
1   France          Perfume
2  Germany      Automobiles
3   Norway  Viking Heritage

In [7]:
europe_df2=europe_df.copy(deep=True)
europe_df2 = europe_df.merge(df_FamousFor)
europe_df2

Country Capital  Population        FamousFor
0    Spain  Madrid      46.770             Wine
1   France   Paris      66.030          Perfume
2  Germany  Berlin      80.620      Automobiles
3   Norway    Oslo       5.084  Viking Heritage

**Note:** Assigning a column that doesn’t exist will create a new column.

#### Summary:
<div class="alert alert-success">
<p style="font-family: Times New Roman; font-size:1.4em;color:Magenta;"> Creating a dataframe and merging the two dataframes.

##### METHOD 3: ADDING A NEW COLUMN TO DATA FRAME USING assign

In [8]:
europe_df4=europe_df.copy(deep=True)
europe_df4.assign(Famousfor=["Wine","Perfume","Automobiles","Viking Heritage"])

Country Capital  Population        Famousfor
0    Spain  Madrid      46.770             Wine
1   France   Paris      66.030          Perfume
2  Germany  Berlin      80.620      Automobiles
3   Norway    Oslo       5.084  Viking Heritage

<strong>Summary: </strong> 
<div class="alert alert-warning">    
<p style="font-family: Times New Roman; font-size:1.4em;color:lime;">    
This method will create a new dataframe with a new column added to the old dataframe.



##### METHOD 4: ADDING A NEW COLUMN TO DATA FRAME 

In [9]:
europe_df1=europe_df.copy(deep=True)
FamousFor = ["Wine","Perfume","Automobiles","Viking Heritage"]
europe_df['FamousFor'] = FamousFor
europe_df

Country Capital  Population        FamousFor
0    Spain  Madrid      46.770             Wine
1   France   Paris      66.030          Perfume
2  Germany  Berlin      80.620      Automobiles
3   Norway    Oslo       5.084  Viking Heritage

#### Summary: 

<div class="alert alert-warning">
<p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
By declaring a new list as a column
<div class="alert alert-warning">
<p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
New column can be added to the dataframe by creating a series(column) inside a dataframe.

### 3. Use the data created in question 1, to do the following: (3 marks)
#### Access column 'Captial' by specifying the column number.
#### Access column 'Population' by specifying the column name.
#### Access a country information

In [10]:
# 1. Access column 'Captial' by specifying the column number.

europe_df.iloc[:,1:2]

Capital
0  Madrid
1   Paris
2  Berlin
3    Oslo

<strong>Summary: </strong> 
<div class="alert alert-danger"> 
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
.iloc[] is primarily integer position based (from 0 to length-1 of the axis), but may also be used with a boolean array.
The first argument ( : ) signifies which rows we would like to index, and the second argument lets us index the column we want. The semicolon returns all of the rows from the column we specified.
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
Allowed inputs are:
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
> An integer, e.g. 5.
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
> A list or array of integers, e.g. [4, 3, 0].
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
> A slice object with ints, e.g. 1:7.
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
> A boolean array.
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
> A callable function with one argument (the calling Series or DataFrame) and that returns valid output for indexing (one of the above). This is useful in method chains, when you don’t have a reference to the calling object, but would like to base your selection on some value.
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
> A tuple of row and column indexes. The tuple elements consist of one of the above inputs, e.g. (0, 1).
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 
.iloc will raise IndexError if a requested indexer is out-of-bounds, except slice indexers which allow out-of-bounds indexing (this conforms with python/numpy slice semantics).

In [11]:
# 2. Access column 'Population' by specifying the column name.

europe_df[["Population"]]

Population
0      46.770
1      66.030
2      80.620
3       5.084

<strong>Summary: </strong> 
<div class="alert alert-success">
<p style="font-family: Times New Roman; font-size:1.4em;color:Magenta;"> Accessing a column by specifying the column name can be simply done by passing the column name in the dataframe.

In [12]:
europe_df.loc[:,'Population']

0    46.770
1    66.030
2    80.620
3     5.084
Name: Population, dtype: float64

#### Summary: 

<div class="alert alert-warning">
<p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
The first argument ( : ) signifies which rows we would like to index, and the second argument (Population) lets us index the column we want. The semicolon returns all of the rows from the column we specified.

In [13]:
# 3. Access a country information
# Lets say we want access information for France

europe_df.iloc[1]

Country        France
Capital         Paris
Population      66.03
FamousFor     Perfume
Name: 1, dtype: object

#### Summary: 
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
Now let’s imagine we needed the information for France. We could simply access it using the iloc function.The above function simply returns the information in row 1. This is useful, but since the data is labeled, we can also use the loc function as well.

In [14]:
europe_df.loc[(europe_df["Country"]=="France")]

Country Capital  Population FamousFor
1  France   Paris       66.03   Perfume

<strong>Summary: </strong> 
<div class="alert alert-warning">    
<p style="font-family: Times New Roman; font-size:1.4em;color:lime;">  
In this case, we are using  simple logic to index our DataFrame: 
we check for all rows where the Name column is France
This will return us a DataFrame matching the result of the iloc 

	
### 4. Read a csv file "example.csv", print it and also check its dimensions ( 2marks)

In [15]:
# Loading the data set

df = pd.read_csv("example.csv")
df

Product ID  Cost Price  Selling Price
0       45SD          60            135
1       12PO          43            121
2       54PL          78            150
3       26PL          65            121
4       68HG          50            132
5       21ER         150            152
6       10FG         132            165
7       57HB         134            161
8       75VB         109            124
9      32HJH         121            152

<strong>Summary: </strong> 
<div class="alert alert-danger"> 
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Orange'> 

Read a comma-separated values (csv) file into DataFrame.

In [16]:
# check dimension/shape of the dataframe

print(f"Dimension is {df.ndim} and shape is {df.shape}")

Dimension is 2 and shape is (10, 3)


#### Summary:
<div class="alert alert-success">
<p style="font-family: Times New Roman; font-size:1.4em;color:olive;">
Attributes of the dataframe-features of an object that describe the object. Does not have paranthesis following them. 
<p style="font-family: Times New Roman; font-size:1.4em;color:olive;">
df.ndim--> gives dimension of the dataframe 
<p style="font-family: Times New Roman; font-size:1.4em;color:olive;">
df.shape--> gives the shape of the dataframe as (rows,columns)
<p style="font-family: Times New Roman; font-size:1.4em;color:olive;">
Dimension of this dataframe is 2 and shape is (10,3) where the dataframe contains 10 rows and 3 columns

### 5. Create a new column BMI, calculate the BMI of persons based on their weight and height. ( 2marks)

In [17]:
height_weight_df = pd.DataFrame({'Name':['Uma','Tamothar','Priyavarshini'],'Height': [1.8,1.7,1.6], 'Weight': [80,90,60]})
height_weight_df['BMI']=height_weight_df.apply(lambda x: (x.Weight/(x.Height**2)), axis=1)
height_weight_df

Name  Height  Weight        BMI
0            Uma     1.8      80  24.691358
1       Tamothar     1.7      90  31.141869
2  Priyavarshini     1.6      60  23.437500

#### Summary: 

<div class="alert alert-warning">
<p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
Step 1: Creating a dataframe with height and weight columns
    <p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
Step 2: Calculating bmi calculation using lambda function and applying across all columns using axis=1 
    <p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
Step 3: Assigning the calculated value to the new column 'BMI'   
    <p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
Step 4: Each person bmi is calculated as Uma=24.6,Tamothar=31.1,Priyavarshini=23.4 respectively. 

### 6. Create a pandas series having values 4, 7, -5, 3, NAN and their index as d, b, a, c, e (2 marks)

In [39]:
value = [4, 7, -5, 3,np.NaN]
ind = ['a', 'b', 'c', 'd','e']
pd_series = pd.Series(value, index=ind)

In [40]:
pd_series



a    4.0
b    7.0
c   -5.0
d    3.0
e    NaN
dtype: float64

#### Summary: 
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
Creating a panda series using pd.Series() with arguments value and index as ind 

### 7. Using the series in question 6, find:                       (2 marks)
#### Find the minimum of all values
#### Find the maximum of all value

In [41]:
# the minimum of all values
pd_series.min()


-5.0

<strong>Summary: </strong> 
<div class="alert alert-warning">    
<p style="font-family: Times New Roman; font-size:1.4em;color:lime;">   
Minimum value from the above series is -5.0

In [42]:
# the maximum of all value
pd_series.max()


7.0

<strong>Summary: </strong> 
<div class="alert alert-danger"> 
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Black'> 
Maximum value from the above series is 7.0

### 8. Using the series in question 6, sort: ( 2 marks)
#### the values in ascending order
#### the values in decending order

In [43]:
pd_series.sort_values(ascending=True)

c   -5.0
d    3.0
a    4.0
b    7.0
e    NaN
dtype: float64

#### Summary:
<div class="alert alert-success">
<p style="font-family: Times New Roman; font-size:1.4em;color:Magenta;">
Sorting the DataFrame by values in the ascending order. So, the series is sorted as -5,3,4,7,NaN

In [44]:
pd_series.sort_values(ascending=False)

b    7.0
a    4.0
d    3.0
c   -5.0
e    NaN
dtype: float64

#### Summary: 

<div class="alert alert-warning">
<p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
Sorting the DataFrame by values in the descending order. So, the series is sorted as 7,4,3,-5,NaN

### 9. Find duplicate rows based on selected column ('Name') ( 2 marks)


**Use the data frame given below:**
 

<table>
<tr>
<th>Name</th>
<th>Salary</th>
<th>City</th>
</tr>
<tr>
<td>John</td>
<td>3400</td>
<td>Sydeny</td>
</tr>
<tr>
<td>Robert</td>
<td>3000</td>
<td>Chicago</td>
</tr>
<tr>
<td>Aadi</td>
<td>1600</td>
<td>New York</td>
</tr>
<tr>
<td>Robert</td>
<td>3000</td>
<td>Chicago</td>
</tr>
<tr>
<td>Robert</td>
<td>3000</td>
<td>Chicago</td>
</tr>
<tr>
<td>Robert</td>
<td>3000</td>
<td>Texas</td>
</tr>
<tr>
<td>Aadi</td>
<td>4000</td>
<td>London</td>
</tr>
<tr>
<td>Sachin</td>
<td>3000</td>
<td>Chicago</td>
</tr>
</table>






In [45]:
data = {
    "Name" : ["John","Robert","Aadi","Robert","Robert","Robert","Aadi","Sachin"],
    "Salary" : [3400,3000,1600,3000,3000,3000,4000,3000],
    "City" : ["Sydney","Chicago","New York","Chicago","Chicago","Texas","London","Chicago"]
}
df = pd.DataFrame(data)
df

Name  Salary      City
0    John    3400    Sydney
1  Robert    3000   Chicago
2    Aadi    1600  New York
3  Robert    3000   Chicago
4  Robert    3000   Chicago
5  Robert    3000     Texas
6    Aadi    4000    London
7  Sachin    3000   Chicago

In [46]:
duplicateRowsDF = df[df.duplicated(['Name'])]
duplicateRowsDF

Name  Salary     City
3  Robert    3000  Chicago
4  Robert    3000  Chicago
5  Robert    3000    Texas
6    Aadi    4000   London

In [47]:
print("Duplicate Rows based on a single column are:", duplicateRowsDF, sep='\n')

Duplicate Rows based on a single column are:
     Name  Salary     City
3  Robert    3000  Chicago
4  Robert    3000  Chicago
5  Robert    3000    Texas
6    Aadi    4000   London


#### Summary: 
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
If we want to compare rows & find duplicates based on selected columns only then we should pass list of column names in
subset argument of the Dataframe.duplicate() function. 
It will select & return duplicate rows based on these passed columns only.
Here rows which has same value in ‘Name’ column are marked as duplicate and returned.

### 10. Get the descriptive statistics of the sales for each season ( 2marks)

**Use the data frame given below:**
   
<table>
<tr>
<th>Month</th>
<th>Sales</th>
<th>Seasons</th>
</tr>
<tr>
<td>Jan</td>
<td>22000</td>
<td>Winter</td>
</tr>
<tr>
<td>Feb</td>
<td>27000</td>
<td>Winter</td>
</tr>
<tr>
<td>Mar</td>
<td>25000</td>
<td>Spring</td>
</tr>
<tr>
<td>Apr</td>
<td>29000</td>
<td>Spring</td>
</tr>
<tr>
<td>May</td>
<td>35000</td>
<td>Spring</td>
</tr>
<tr>
<td>June</td>
<td>67000</td>
<td>Summer</td>
</tr>
<tr>
<td>July</td>
<td>78000</td>
<td>Summer</td>
</tr>
<tr>
<td>Aug</td>
<td>67000</td>
<td>Summer</td>
</tr>
<tr>
<td>Sep</td>
<td>56000</td>
<td>Fall</td>
</tr>
<tr>
<td>Oct</td>
<td>56000</td>
<td>Fall</td>
</tr>
<tr>
<td>Nov</td>
<td>56000</td>
<td>Fall</td>
</tr>
<tr>
<td>Dec</td>
<td>60000</td>
<td>Winter</td>
</tr>

</table>

In [48]:
data = {
    "Month" : ["Jan","Feb","Mar","Apr","May","June","July","Aug","Sep","Oct","Nov","Dec"],
    "Sales" : [22000,27000,25000,29000,35000,67000,78000,67000,56000,56000,56000,60000],
    "Seasons" : ["Winter","Winter","Spring","Spring","Spring","Summer","Summer","Summer","Fall","Fall","Fall","Winter"]
}
df = pd.DataFrame(data)
df.T

0       1       2       3       4       5       6       7   \
Month       Jan     Feb     Mar     Apr     May    June    July     Aug   
Sales     22000   27000   25000   29000   35000   67000   78000   67000   
Seasons  Winter  Winter  Spring  Spring  Spring  Summer  Summer  Summer   

            8      9      10      11  
Month      Sep    Oct    Nov     Dec  
Sales    56000  56000  56000   60000  
Seasons   Fall   Fall   Fall  Winter

In [49]:
df[df['Seasons']=='Fall'].describe().T

count     mean  std      min      25%      50%      75%      max
Sales    3.0  56000.0  0.0  56000.0  56000.0  56000.0  56000.0  56000.0

In [50]:
df[df['Seasons']=='Spring'].describe().T

count          mean          std      min      25%      50%      75%  \
Sales    3.0  29666.666667  5033.222957  25000.0  27000.0  29000.0  32000.0   

           max  
Sales  35000.0

In [51]:
df[df['Seasons']=='Summer'].describe().T

count          mean          std      min      25%      50%      75%  \
Sales    3.0  70666.666667  6350.852961  67000.0  67000.0  67000.0  72500.0   

           max  
Sales  78000.0

In [52]:
df[df['Seasons']=='Winter'].describe().T

count          mean           std      min      25%      50%      75%  \
Sales    3.0  36333.333333  20647.840888  22000.0  24500.0  27000.0  43500.0   

           max  
Sales  60000.0

<strong>Summary: </strong> 
<div class="alert alert-warning">    
<p style="font-family: Times New Roman; font-size:1.4em;color:lime;">   
We are using conditional statement to filter out a particular rows from the dataframe. Then we use describe() function 
to give descriptive statistics of sales for each season

In [53]:
df.groupby('Seasons')['Sales'].agg([np.mean, np.std,np.min,np.median,np.var,np.sum,np.max])

mean           std   amin   median           var     sum  \
Seasons                                                                     
Fall     56000.000000      0.000000  56000  56000.0  0.000000e+00  168000   
Spring   29666.666667   5033.222957  25000  29000.0  2.533333e+07   89000   
Summer   70666.666667   6350.852961  67000  67000.0  4.033333e+07  212000   
Winter   36333.333333  20647.840888  22000  27000.0  4.263333e+08  109000   

          amax  
Seasons         
Fall     56000  
Spring   35000  
Summer   78000  
Winter   60000

<strong>Summary: </strong> 
<div class="alert alert-danger"> 
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Gray'> 
df.groupby('Seasons') splits the data into different groups, based on the Seasons column
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Gray'> 
['Sales'] selects only that column from the groupings
<p style="font-family: Times New Roman; font-size:1.4em;color:gold;"><span style='color:Gray'> 
.agg() method allows us to easily generate summary statistics based on our different groups

### 11. Combine the new column age in the below data frame (2 marks)

**Use the data frame given below:**
   
<table>
<tr>
<th>Name</th>
<th>Maths</th>
<th>Science</th>
<th>English</th>
</tr>
<tr>
<td>Emma</td>
<td>56</td>
<td>89</td>
<td>89</td>
</tr>
<tr>
<td>Mia</td>
<td>78</td>
<td>87</td>
<td>89</td>
</tr>
<tr>
<td>Sophia</td>
<td>78</td>
<td>78</td>
<td>76</td>
</tr>
<tr>
<td>James</td>
<td>67</td>
<td>89</td>
<td>78</td>
</tr>
<td>John</td>
<td>88</td>
<td>78</td>
<td>87</td>
</tr>
   
</table>

In [54]:
data={
    'Name': ['Emma', 'Mia', 'Sophia', 'James', 'John'],
    'Maths': [56, 78, 78, 67, 88],
    'Science': [89, 87, 78, 89, 78],
    'English': [89, 89, 76, 78, 87],
}
df = pd.DataFrame(data)
df

Name  Maths  Science  English
0    Emma     56       89       89
1     Mia     78       87       89
2  Sophia     78       78       76
3   James     67       89       78
4    John     88       78       87

In [55]:
age={'Emma':16,'Mia':15,'Sophia':14,'James':13,'John':10}
df['Age']=df.Name.map(age)
df

Name  Maths  Science  English  Age
0    Emma     56       89       89   16
1     Mia     78       87       89   15
2  Sophia     78       78       76   14
3   James     67       89       78   13
4    John     88       78       87   10

#### Summary:
<div class="alert alert-success">
<p style="font-family: Times New Roman; font-size:1.4em;color:Magenta;">
Creating a dictionary with Name as keys and Age as items. Then mapping age to 'Age' column

In [56]:
data = {
    'Name': ['Emma', 'Mia', 'Sophia', 'James', 'John'],
    'Maths': [56, 78, 78, 67, 88],
    'Science': [89, 87, 78, 89, 78],
    'English': [89, 89, 76, 78, 87],
}
df = pd.DataFrame(data)
dataage = {
    'Name': ['Emma', 'Mia', 'Sophia', 'James', 'John'],
    'Age': [21, 24, 40, 32, 43]
}
dfAge = pd.DataFrame(dataage)

In [57]:
df = df.merge(dfAge)
df

Name  Maths  Science  English  Age
0    Emma     56       89       89   21
1     Mia     78       87       89   24
2  Sophia     78       78       76   40
3   James     67       89       78   32
4    John     88       78       87   43

#### Summary: 

<div class="alert alert-warning">
<p style="font-family: Times New Roman; font-size:1.4em;color:brown;">
Combining a new column Age using merge function

### 12.Concatenate two data frames along with the columns

**Use the data frame given below:**
   
<table>
<tr>
<th>ID</th>
<th>Name</th>
<th>Subject</th>
</tr>
<tr>
<td>101</td>
<td>Alex</td>
<td>Maths</td>
</tr>
<tr>
<td>102</td>
<td>Amy</td>
<td>English</td>
</tr>
<tr>
<td>103</td>
<td>Allen</td>
<td>Science</td>
</tr>
<tr>
<td>104</td>
<td>Alice</td>
<td>German</td>
</tr>
<td>105</td>
<td>Ayoung</td>
<td>History</td>
</tr>
   
</table><br>



<table>
<tr>
<th>ID</th>
<th>Name</th>
<th>Subject</th>
</tr>
<tr>
<td>101</td>
<td>Billy</td>
<td>English</td>
</tr>
<tr>
<td>102</td>
<td>Brian</td>
<td>Science</td>
</tr>
<tr>
<td>103</td>
<td>Bran</td>
<td>Social Science</td>
</tr>
<tr>
<td>104</td>
<td>Bryce</td>
<td>German</td>
</tr>
<td>105</td>
<td>Betty</td>
<td>History</td>
</tr>
   
</table> 

In [58]:
data1 = {
    'ID': [101, 102, 103, 104, 105],
    'Name': ['Alex', 'Amy', 'Allen', 'Alice', 'Ayoung'],
    'Subject': ['Maths', 'English', 'Science', 'German', 'History'],
}
df1 = pd.DataFrame(data1)
data2 = {
    'ID': [101, 102, 103, 104, 105],
    'Name': ['Billy', 'Brain', 'Bran', 'Bryce', 'Betty'],
    'Subject': ['Maths', 'English', 'Science', 'German', 'History']
}
df2 = pd.DataFrame(data2)

In [59]:
data = pd.concat([df1,df2],axis="columns") #Concatenate along the column axis
data

ID    Name  Subject   ID   Name  Subject
0  101    Alex    Maths  101  Billy    Maths
1  102     Amy  English  102  Brain  English
2  103   Allen  Science  103   Bran  Science
3  104   Alice   German  104  Bryce   German
4  105  Ayoung  History  105  Betty  History

#### Summary: 
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
●The concat() method can be used to concatenate more than two DataFrames simultaneously.
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
●By default, the concat() method concatenates along the axis = 0(vertically)  
data3 = pd.concat([df1,df2]) #Concatenate along the row axis by default
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
●The concatenation is in the order they are passed in the function
<div class="alert alert-danger"><p style="font-family: Times New Roman; font-size:1.4em;color:Purple;">
●The index numbers of the concatenated DataFrame are of the actual DataFrames.
